controllers/progresses.js

```diff
@@ -107,8 +108,38 @@ const createProgress = async (req, res) => {
  */
 
 const getProgress = async (req, res) => {
+  const { dev, page = 0, size = 100, type, userId, taskId } = req.query;
   try {
     const data = await getProgressDocument(req.query);
+    if (dev) {
+      const paginatedProgressDocs = await getPaginatedProgressDocument(req.query);
+      const limit = parseInt(size, 10);
+      const offset = parseInt(page, 10) * limit;
+      const total = data.length;
+      const nextPage = offset + limit < total ? parseInt(page, 10) + 1 : null;
+      const prevPage = page > 0 ? parseInt(page, 10) - 1 : null;
+
+      let baseUrl = `${req.baseUrl}`;
```

Comment:
Do we need to do this?

Reply:
Yes, we need this to calculate the prev and next page links in the API response. These values help us determine when to set them as null if there are no previous or next pages available.